rust/kernel/user_ptr.rs

```diff
@@ -82,29 +79,14 @@ pub struct UserSlicePtr(*mut c_types::c_void, usize);
 impl UserSlicePtr {
     /// Constructs a user slice from a raw pointer and a length in bytes.
     ///
-    /// Checks that the provided range is within the legal area for
-    /// userspace memory, using `access_ok` (e.g., on i386, the range
-    /// must be within the first 3 GiB), but does not check that
-    /// the actual pages are mapped in the current process with
-    /// appropriate permissions. Those checks are handled in the read
-    /// and write methods.
-    ///
     /// # Safety
     ///
-    /// This is `unsafe` because if it is called within `set_fs(KERNEL_DS)`
-    /// context then `access_ok` will not do anything. As a result the only
-    /// place you can safely use this is with a `__user` pointer that was
-    /// provided by the kernel.
-    ///
```

Comment:
I think this paragraph is basically still true, even if we're not calling `access_ok` here, the function is unsafe because if you use this type in a `set_fs(KERNEL_DS)` context, bad things will happen.

Reply:
gregkh also tells me that `set_fs` is not a thing anymore, it only exists for specific arch-specific things, and that we should ignore it.

Reply:
set_fs() is going away, yeah, but the concept of checking whether something is a valid userspace pointer is, as well? (i.e. access_ok()?)

Reply:
I'm not sure if it's going away entirely, but callers of `copy_from_user` and `copy_to_user` should not call it. gregkh says this has been the case for years.

Reply:
IMO it's a bit easier to reason about the security properties with that invariant, but if it's really not necessary, I suppose its fine.

Reply:
So we can remove the paragraph because reads/writes through the type will always call `copy_{from,to}_user()` and those are supposed to be OK without a check before, right? Perhaps we can state this somewhere.

Reply:
I mean, the type says the calls go through there, but it doesn't explain the fact that calling those is enough now.

Reply:
How is a developer going to determine whether a range is valid or not before calling `new`?

Saying that we got it from the 'kernel' is not sufficient. We have an example in `ioctl`: although it 'comes from the kernel', it is just a plain untrusted arbitrary value passed in by userspace.